src/api/services/WorkflowService.ts

```
@@ -55,6 +55,18 @@ export class WorkflowService implements IWorkflowService {
         return externalWorkflowService.completeTask(id, mindstrongUser);
     }
 
+    public async updateTask(id: string, taskUpdatePayload: TaskUpdatePayload, mindstrongUser: MindstrongUser) {
+        const task: TaskDTO = await externalWorkflowService.getTask(id, mindstrongUser);
+
+        if (task.assignee && task.assignee._id != mindstrongUser.mindstrongUserID) {
```

if the task has no assignee, we allow to update (this might be ok, but want to make sure its intentional)?

---

For the MVP at least, a task should always have assignee. In case it doesn't for some reason, it won't be visible to anyone. In that case, it might be ok to allow this to any valid caseworker in the backend. It will still verify the patient's (if exists on the task) affiliation against the requesting caseworker's affiliations.

---

let's for now raise an exception if there is no assignee as it should be an unexpected error case and we can relax this later when the intention is there

---

added an exception for no assignee